server.js

```diff
@@ -765,12 +766,44 @@ cache(function(data, match, sendBadge, request) {
     }
     try {
       var data = JSON.parse(buffer);
-      // Grab the latest stable version, or an unstable
+
       var versions = Object.keys(data.package.versions);
-      var version = latestVersion(versions);
-      var vdata = versionColor(version);
-      badgeData.text[1] = vdata.version;
-      badgeData.colorscheme = vdata.color;
+      var stableVersion = latestVersion(versions);
+      var vdata = versionColor(stableVersion);
+
+      var badgeText = null;
+      var badgeColor = null;
+
+      switch (info) {
+          case 'v':
+              badgeText = vdata.version;
+              badgeColor = vdata.color;
+              break;
+          case 'vpre':
+              if (versions[0] !== undefined) {
+
+                  var unstableVersion = null;
+                  for (var i = 0; i < versions.length; i++) {
+                      var version = versions[i];
+
+                      if (version.charAt(0) !== 'v') {
```

Comment:
What if the version does start with a `v`? Do we never match a version at all?

As a side-note, `version[0] !== 'v'` sounds like what you mean; `charAt()` returns an integer, which will never match a string.